packages/core/src/common/keybinding.ts

```diff
@@ -53,6 +59,88 @@ export namespace KeybindingContexts {
     };
 }
 
+export class KeybindingIndex {
+    private bindings: Keybinding[] = [];
+
+    constructor(private parent?: KeybindingIndex) { }
+
+    getKeybindingsForCommand(command: string): Keybinding[] {
+        const result: Keybinding[] = [];
+
+        if (this.parent) {
+            for (const parentBinding of this.parent.getKeybindingsForCommand(command)) {
+                if (!this.isShadowed(parentBinding)) {
+                    result.push(parentBinding);
+                }
+            }
+        }
+
+        for (const binding of this.bindings) {
+            if (binding.commandId === command) {
+                result.push(binding);
+            }
+        }
+        return result;
+    }
+
+    getKeyBindingsForKeyCode(keyCode: KeyCode): Keybinding[] {
+        const result: Keybinding[] = [];
+
+        if (this.parent) {
+            for (const parentBinding of this.parent.getKeyBindingsForKeyCode(keyCode)) {
+                if (!this.isShadowed(parentBinding)) {
+                    result.push(parentBinding);
+                }
+            }
+        }
+
+        for (const binding of this.bindings) {
+            if (TheiaKeyCodeUtils.equals(binding.keyCode, keyCode)) {
+                result.push(binding);
+            }
+        }
+        return result;
+    }
+
+    protected isShadowed(fromParent: Keybinding): boolean {
+        for (const binding of this.bindings) {
+            if (fromParent.commandId === binding.commandId
+                || TheiaKeyCodeUtils.equals(fromParent.keyCode, binding.keyCode)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    setParentKeyBinding(keybinding: Keybinding): void {
```

Comment:
We should just keep a handle to the parent and remove these two 'parent'-methods

Reply:
This is basically just setting the keybindings from the contributions, you mean maybe the name isn't correct? Basically there's already a handle to the parent here
```
constructor(private parent?: KeybindingIndex) 
```

Unless you mean something else?

Reply:
I meant the only client to these methods is below (see my comment). And if we instead use the defaultKeybindingIndex we don't need these methods at all.